docs/deploying/ssl.html

```diff
@@ -0,0 +1,114 @@
+          <li>Install OpenSSL: In this guide, we will use OpenSSL to generate our self-signed certificate.</li>
+      </ul>
+      <p>OpenSSL is included by default on Linux. On MacOS we can install it using Homebrew:</p>
+      <pre><code>brew install openssl</code></pre>
+      <p>We then link it so we can use the OpenSSL CLI:</p>
+      <pre><code>ln -s /usr/local/opt/openssl/include/openssl /usr/local/include</code></pre>
+      <div class="underline"></div>
+      <h2 class="heading-2"><span class="blue-text">Step 1:</span> Create our certificate and key pair.</h2>
+      <p class="block-text">To enable TLS, a server needs a certificate and a corresponding secret key.
+           Certificates are files that bind together information about the identity of the owner of a site and the public half of an asymmetric key pair (usually RSA).
+           Certificates are usually digitally signed by a certificate authority (CA) who verifies that the identity information in the certificate is correct.
+           This creates a chain of certificates between the site owner certificate and a CA certificate and transitive trust.
+            Assuming that we trust the CA, we can trust the validity of the server certificate.</p>
+       <div class="info">
+          <p class="block-text">If you want to create a CA-signed certificate chain, a useful tool to use is <a href="https://letsencrypt.org/" target="_blank">Let’s Encrypt</a>,
+           a free, automated and open CA, provided by the Internet Security Research Group (ISRG).</p>
+       </div>
+      <p>For this guide we will use OpenSSL to generate a self-signed certificate.</p>
+      <p>Create a Credentials folder in the root of your project:</p>
+      <pre><code>mkdir Credentials && cd Credentials</code></pre>
+      <p>Inside this folder we create a 2048 bit RSA private key using OpenSSL:</p>
+      <pre><code>openssl genrsa -out key.pem 2048</code></pre>
+      <p>We use this private key to create a certificate:</p>
+      <pre><code>openssl req -new -sha256 -key key.pem -out csr.csr</code></pre>
+      <p class="block-text">At this stage you provide additional information that will be stored in the certificate.</p>
+      <p class="block-text">These choices are optional except for Common Name which must be the servers hostname. In our case this is "localhost".</p>
+      <p>Next we use OpenSSL to convert the certificate to PEM format:</p>
+      <pre><code>openssl req -x509 -sha256 -days 365 -key key.pem -in csr.csr -out cert.pem</code></pre>
+      <p>Finally we encrypt the key and certificate together using a password. For simplicity we will use "password".</p>
+      <pre><code>openssl pkcs12 -export -out cert.pfx -inkey key.pem -in cert.pem</code></pre>
+<div class="underline"></div>
+      <h2 class="heading-2"><span class="blue-text">Step 2:</span> Setting up your SSL configuration</h2>
+      <p>Open your `Sources` > `Application` > `Application.swift`</p>
```

Comment:
In an ideal world it is always nicer when we explain what we're going to do in the step, before we ask users to start opening files.